### PR TITLE
 feat(header): add download app to menu and use new `<Icon />`

### PR DESCRIPTION
**Pull request by CedricPoon-SeekAsia**
As stated in [here](https://trello.com/c/YL5LyFr0/195-add-download-app-in-navigation-menu) and [here](https://trello.com/c/pl5l96Tn/410-bug-missing-icons-on-the-menu-in-mobile-view), new `<Icon>` will replace all old type icons. A new menu list download app is also added.

### CODE REVIEW

File: jobsDB/Header/Header.js [hunk 1]
```diff
@@ -3,53 +3,63 @@ import PropTypes from 'prop-types';
...
     url: baseUrl + messages['header.myJobsDBJobAlertsListUrl']
   },
   {
     title: messages['header.myJobsDBSaveJobsTitle'],
-    ItemIcon: IconBookmark,
+    ItemIcon: <Icon type='bookmark' lineHeight='conventional' size='small' className={styles.menuIcon} />,
     url: messages['header.myJobsDBSaveJobsUrl']
   }];
 
+  const downloadItems = [
```
Comment (mengtzu) [inline jobsDB/Header/Header.js]: 🤔 normally I don't like browser sniffing but this feels like a legit case (or OS sniffing I guess).  What would we think about a dirty OS check to add one link or the other?

File: react/ListItem/__snapshots__/ListItem.test.js.snap [hunk 2]
```diff
@@ -3,6 +3,7 @@
 exports[`ListItem: should render custom left padding 1`] = `
 <div
   className="root"
+  leftPadding={8}
```
Comment (ChanJit) [inline react/ListItem/__snapshots__/ListItem.test.js.snap]: why the snap shot end up like that? hmmm why have `leftPadding`?

Comment (CedricPoon-SeekAsia) [reply react/ListItem/__snapshots__/ListItem.test.js.snap]: @ChanJit It should be the original feature of `<ListItem />` cause there may be a second level sublist in a list.

Comment (chiwongseekasia) [reply react/ListItem/__snapshots__/ListItem.test.js.snap]: cause he added restprops on listitem for passing onclick func in.

File: react/Header/components/Menu/Menu.js [hunk 3]
```diff
@@ -12,8 +12,17 @@ export default class Menu extends Component {
 
     this.state = {
       subMenuOpen: false,
+      subMenuOpenIndex: 0,
       localesMenuOpen: false
     };
+
+    this.openSubMenu = this.openSubMenu.bind(this);
+    this.toggleSubMenu = this.toggleSubMenu.bind(this);
+    this.toggleLocalesMenu = this.toggleLocalesMenu.bind(this);
```
Comment (cm-seekasia) [inline react/Header/components/Menu/Menu.js]: Should we remove those `bind` and change the func like:
```
openSubMenu = i => {
    this.setState({ subMenuOpenIndex: i });
  }
```

File: react/Header/components/Menu/Menu.js [hunk 4]
```diff
@@ -65,26 +87,27 @@ export default class Menu extends Component {
...
+              {messages['header.profileTitle']}
             </MenuItem>
-            <MenuItem linkUrl={baseUrl + messages['header.invitationUrl']} ItemIcon={JobInvitationIcon} brandStyles={brandStyles} linkRenderer={linkRenderer} >
-              <Text>{messages['header.invitationTitle']}</Text>
+            <MenuItem linkUrl={baseUrl + messages['header.invitationUrl']} ItemIcon={<Icon type='jobInvitation' lineHeight='conventional' size='small' className={brandStyles.menuIcon} />} linkRenderer={linkRenderer} >
+              {messages['header.invitationTitle']}
             </MenuItem>
           </div>)
         }
         {this.renderMenuLinks({ messages, brandStyles, linkRenderer }, links)}
```
Comment (cm-seekasia) [inline react/Header/components/Menu/Menu.js]: change to `this.renderMenuLinks({ messages, brandStyles, linkRenderer, links })` is better?

**Comment by cm-seekasia** [comment]
The icon size, font style and alignment are different now. The old one is better?

Old:
image
New:
image

**Comment by chiwongseekasia** [comment]
> The icon size, font style and alignment are different now. The old one is better?
> 
> Old:
> image
> New:
> image

@CedricPoon-SeekAsia 
need to align lineheight in icon component  and text component next to it

File: jobsDB/Header/Header.js [hunk 5]
```diff
@@ -3,53 +3,63 @@ import PropTypes from 'prop-types';
...
 
-  let icon = LightbulbIcon;
+  let icon = <Icon type='careerInsight' lineHeight='conventional' size='small' className={styles.menuIcon} />;
 
   if (country === 'th') {
-    icon = EducationIcon;
+    icon = <Icon type='education' lineHeight='conventional' size='small' className={styles.menuIcon} />;
   } else if (country === 'id') {
-    icon = EventIcon;
+    icon = <Icon type='calendar' lineHeight='conventional' size='small' className={styles.menuIcon} />;
```
Comment (ashin) [inline jobsDB/Header/Header.js]: Can we have this `if/else` logic only setting the Icon type prop? That will make it super easy to figure out what is changing with different countries.

File: react/Header/components/Menu/Menu.js (hunk 3 above)
Comment (ashin) [reply react/Header/components/Menu/Menu.js]: I don't mind either way. Although it's probably best to use which ever standard we have used throughout the rest of this application.

File: react/Header/components/Menu/components/MenuItem/MenuItem.less [hunk 6]
```diff
@@ -6,6 +6,10 @@
   border-bottom: 1px solid @saGrey5;
```
Comment (ashin) [inline react/Header/components/Menu/components/MenuItem/MenuItem.less]: This seems to inherit padding from `ListItem` component, which only has `10px` of left padding, but the heading text has `16px`.

If we set a custom `paddingLeft: @gridbase * 4` here than the icons should left align to the heading like before.

File: react/ListItem/__snapshots__/ListItem.test.js.snap (hunk 2 above)
Comment (CedricPoon-SeekAsia) [reply react/ListItem/__snapshots__/ListItem.test.js.snap]: @ChanJit I pass `onClick` explicitly into `<ListItem>`, should now be okay 😃

File: react/Header/components/Menu/components/MenuItem/MenuItem.less (hunk 6 above)
Comment (CedricPoon-SeekAsia) [reply react/Header/components/Menu/components/MenuItem/MenuItem.less]: @ashin new feature of `<Icon>` component will maintain the left spacing in `<ListItem>`. Will be handled by @patrickliu-sa

File: react/Header/components/Menu/Menu.js (hunk 3 above)
Comment (CedricPoon-SeekAsia) [reply react/Header/components/Menu/Menu.js]: @cm-seekasia @ashin will keep

File: jobsDB/localization/en-id.js [hunk 7]
```diff
@@ -22,6 +22,11 @@ export default {
   'header.careerInsightsUrl': '/id-id/pages/event',
   'header.employerSiteTitle': 'For Employers',
   'header.employerSiteUrl': '/id/en/home/employerindex',
+  'header.downloadApp': 'Download Our App',
+  'header.playStore': 'Google Play Store',
+  'header.playStoreUrl': 'https://play.google.com/store/apps/details?id=com.jobsdb&utm_campaign=MobileNavigationMenuAppDownload&utm_source=MobileNavigationMenu&utm_medium=banner',
+  'header.appStore': 'App Store',
+  'header.appStoreUrl': 'https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432?utm_campaign=MobileNavigationMenuAppDownload&utm_source=MobileNavigationMenu&utm_medium=banner',
```
[the remark this answers is not in the document]
Comment (CedricPoon-SeekAsia) [reply jobsDB/localization/en-id.js]: @cm-seekasia updated in latest commit

File: jobsDB/localization/en-th.js [hunk 8]
```diff
@@ -21,6 +21,11 @@ export default {
   'header.careerInsightsUrl': '/th-th/pages/jobsdb-campnet',
   'header.employerSiteTitle': 'For Employers',
   'header.employerSiteUrl': '/th/en/home/employerindex',
+  'header.downloadApp': 'Download Our App',
+  'header.playStore': 'Google Play Store',
+  'header.playStoreUrl': 'https://play.google.com/store/apps/details?id=com.jobsdb&utm_campaign=MobileNavigationMenuAppDownload&utm_source=MobileNavigationMenu&utm_medium=banner',
+  'header.appStore': 'App Store',
+  'header.appStoreUrl': 'https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432?utm_campaign=MobileNavigationMenuAppDownload&utm_source=MobileNavigationMenu&utm_medium=banner',
```
[the remark this answers is not in the document]
Comment (CedricPoon-SeekAsia) [reply jobsDB/localization/en-th.js]: @cm-seekasia updated in latest commit

File: jobsDB/localization/id-id.js [hunk 9]
```diff
@@ -21,6 +21,11 @@ export default {
   'header.careerInsightsUrl': '/id-id/pages/event',
   'header.employerSiteTitle': 'Situs Perusahaan',
   'header.employerSiteUrl': '/id/id/home/employerindex',
+  'header.downloadApp': 'Download Our App',
+  'header.playStore': 'Google Play Store',
+  'header.playStoreUrl': 'https://play.google.com/store/apps/details?id=com.jobsdb&utm_campaign=MobileNavigationMenuAppDownload&utm_source=MobileNavigationMenu&utm_medium=banner',
+  'header.appStore': 'App Store',
+  'header.appStoreUrl': 'https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432?utm_campaign=MobileNavigationMenuAppDownload&utm_source=MobileNavigationMenu&utm_medium=banner',
```
[the remark this answers is not in the document]
Comment (CedricPoon-SeekAsia) [reply jobsDB/localization/id-id.js]: @cm-seekasia updated in latest commit

File: react/Header/components/Menu/components/MenuItem/MenuItem.less (hunk 6 above)
Comment (chiwongseekasia) [reply react/Header/components/Menu/components/MenuItem/MenuItem.less]: @ashin and @cm-seekasia 
I think this should open another PR for it.

Comment (ashin) [reply react/Header/components/Menu/components/MenuItem/MenuItem.less]: Ahhhh, nice. Okay then I'm fine with this going ahead.

**Pull request merged**